packages/utils/typescript/lib/generators/schemas/attributes.js

```diff
@@ -0,0 +1,241 @@
+'use strict';
+
+const { factory } = require('typescript');
+const fp = require('lodash/fp');
+
+const { addImport } = require('./imports');
+const { getTypeNode, toTypeLitteral } = require('./utils');
+
+/**
+ * Generate a property signature node for a given attribute
+ *
+ * @param {object} schema
+ * @param {string} attributeName
+ * @param {object} attribute
+ * @returns {object}
+ */
+const attributeToPropertySignature = (schema, attributeName, attribute) => {
+  const baseType = getAttributeType(attributeName, attribute, schema.uid);
+  const modifiers = getAttributeModifiers(attributeName, attribute);
+
+  const nodes = [baseType, ...modifiers];
+
+  return factory.createPropertySignature(
+    undefined,
+    factory.createIdentifier(attributeName),
+    undefined,
+    factory.createIntersectionTypeNode(nodes)
+  );
+};
+
+/**
+ * Create the base type node for a given attribute
+ *
+ * @param {string} attributeName
+ * @param {object} attribute
+ * @param {string} uid
+ * @returns {object}
+ */
+const getAttributeType = (attributeName, attribute, uid) => {
+  const mappers = {
```

Comment:
you should put your mappers outside of the function they don't need to be redefined on every call